Code/Engine/RendererDX11/Device/Implementation/DeviceDX11.cpp

```diff
@@ -691,6 +692,62 @@ void ezGALDeviceDX11::DestroyVertexDeclarationPlatform(ezGALVertexDeclaration* p
   EZ_DELETE(&m_Allocator, pVertexDeclarationDX11);
 }
 
+void ezGALDeviceDX11::UpdateBufferForNextFramePlatform(const ezGALBuffer* pBuffer, ezConstByteArrayPtr sourceData, ezUInt32 uiDestOffset)
+{
+  const ezGALBufferDX11* pBufferDX11 = static_cast<const ezGALBufferDX11*>(pBuffer);
+
+#if EZ_ENABLED(EZ_COMPILE_FOR_DEBUG)
+  for (auto& copy : m_PendingCopies)
+  {
+    if (copy.m_pDestResource == pBufferDX11->GetDXBuffer())
```

Comment:
Can we allow updating the same buffer twice in a frame if the ranges don't overlap? For larger structured buffers it would be beneficial to only update parts of the buffer, which aren't necesarily in one block. If we don't have a use case for that yet, we can also extend it in a later PR?